messaging/src/main/java/org/axonframework/eventhandling/pooled/PooledStreamingEventProcessor.java

```diff
@@ -442,7 +443,7 @@ public static class Builder extends AbstractEventProcessor.Builder {
         private Function<StreamableMessageSource<TrackedEventMessage<?>>, TrackingToken> initialToken =
                 ms -> ReplayToken.createReplayToken(ms.createHeadToken());
         private long tokenClaimInterval = 5000;
-        private int maxClaimedSegments = Short.MAX_VALUE;
+        private MaxSegmentProvider maxSegmentProvider;
```

Comment:
This could default to `p -> Short.MAX_VALUE`

Reply:
Resolved.